lib/rules/prefer-promise-strategies.js

```diff
@@ -0,0 +1,47 @@
+'use strict'
+
+// Matches X.{and,withArgs(Y)}.returnValue(Z)
+const SpyStrategyCall = `
+  CallExpression:matches(
+    [callee.object.property.name=and],
+    [callee.object.callee.property.name=withArgs]
+  )
+`.replace(/\s+/g, ' ').trim()
+const ReturnStrategyCtor = `${SpyStrategyCall}[callee.property.name=returnValue]`
+
+// Matches Promise.{resolve,reject}(X)
+const PromiseCall = 'CallExpression[callee.object.name=Promise]'
+const SettledPromiseCtor = `${PromiseCall}[callee.property.name=/resolve|reject/]`
+
+module.exports = function (context) {
+  return {
+    // TODO: Change to "FOO:has(> BAR)" once available in esquery
```

Comment:
If you are planning to leave this comment. Please add a link to esquery issue/ discussion/ anything that helps tracking the completion of this feature.

Reply:
Will do

Reply:
I removed it, since it does not really make a difference